### PR TITLE
Open Source Grafbase CLI

### PR DESCRIPTION
**Pull request by yoav-lavi**
# Description

## Features

- Open sources the Grafbase CLI 🎉

# Type of change

- [ ] 💔 Breaking
- [x] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,5 @@
+root = true
```
Comment (fbjork) [inline .editorconfig:1]: ignore this file?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# These owners will be the default owners for everything in the repo. Unless a
+
+# later match takes precedence, the team will be requested for review when
+
+# someone opens a pull request.
+
+cli/ @grafbase/backend
```
Comment (fbjork) [inline .github/CODEOWNERS:7]: should we create a `cli` team that could include outside collaborators eventually?

File: .github/disabled-workflows/cli-coverage.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+name: CLI
```
Comment (fbjork) [inline .github/disabled-workflows/cli-coverage.yml:1]: Why is this disabled?

File: .github/workflows/cli-build.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+name: CLI
```
Comment (fbjork) [inline .github/workflows/cli-build.yml:1]: Should we give this a more specific name, e.g `CLI Build`?

File: cli/crates/cli/package.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "grafbase",
+  "version": "0.1.0",
+  "description": "placeholder",
```
Comment (fbjork) [inline cli/crates/cli/package.json:4]:
```suggestion
  "description": "The data platform for developers.",
```

File: cli/crates/dev-server/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,44 @@
+[package]
+name = "grafbase-local-dev-server"
```
Comment (fbjork) [inline cli/crates/dev-server/Cargo.toml:2]: What if we used the name `grafbase-dev-server`. `dev-server` implies it's local IMO.

File: cli/README.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
+<div align="center">
```
Comment (fbjork) [inline cli/README.md:1]: I would use the same styling as the top level README here:

```
<p align="center">
  <a href="https://grafbase.com">
    <img src="https://grafbase.com/images/other/grafbase-logo-circle.png" height="96">
    <h3 align="center">Grafbase CLI</h3>
  </a>
</p>
```

File: README.md [hunk 8]
```diff
@@ -9,11 +9,18 @@
   The data platform for developers.
 </p>
 
+<p align="center">
+  <a href="https://github.com/grafbase/grafbase/actions/workflows/cli-build.yml">
+    <img alt="CLI build status" src=https://github.com/grafbase/grafbase/actions/workflows/cli-build.yml/badge.svg>
```
Comment (fbjork) [inline README.md:14]: Should we perhaps keep this README clean and only show the badge in the README of the CLI?

File: .editorconfig (hunk 1 above)
Comment (yoav-lavi) [reply .editorconfig:1]: This is meant to be committed, it enforces a uniform formatting for all files (like prettier, but for files prettier doesn't format) https://editorconfig.org

File: .github/CODEOWNERS (hunk 2 above)
Comment (yoav-lavi) [reply .github/CODEOWNERS:7]: We could yeah, but for review purposes I think backend makes sense no?

File: cli/crates/dev-server/Cargo.toml (hunk 6 above)
Comment (yoav-lavi) [reply cli/crates/dev-server/Cargo.toml:2]: I was keeping the same convention, we could go with either that or `grafbase-local-server`

File: .editorconfig (hunk 1 above)
Comment (yoav-lavi) [reply .editorconfig:1]: It was in the CLI repo beforehand

**Review by fbjork — APPROVED** [review]
Amazing work!

File: .github/workflows/cli-build.yml (hunk 4 above)
Comment (fbjork) [reply .github/workflows/cli-build.yml:1]: I see. I guess we can keep it as is then.

File: .github/disabled-workflows/cli-coverage.yml (hunk 3 above)
Comment (fbjork) [reply .github/disabled-workflows/cli-coverage.yml:1]: Ok, remove the file for now then?

File: .editorconfig (hunk 1 above)
Comment (fbjork) [reply .editorconfig:1]: Got it

File: cli/crates/dev-server/Cargo.toml (hunk 6 above)
Comment (fbjork) [reply cli/crates/dev-server/Cargo.toml:2]: `grafbase-local-server` works

File: .github/disabled-workflows/cli-coverage.yml (hunk 3 above)
Comment (yoav-lavi) [reply .github/disabled-workflows/cli-coverage.yml]: Done

File: .github/workflows/cli-build.yml (hunk 4 above)
Comment (yoav-lavi) [reply .github/workflows/cli-build.yml]: Added specific naming for now

File: cli/crates/dev-server/Cargo.toml (hunk 6 above)
Comment (yoav-lavi) [reply cli/crates/dev-server/Cargo.toml]: Done

File: cli/README.md (hunk 7 above)
Comment (yoav-lavi) [reply cli/README.md]: Done

File: README.md (hunk 8 above)
Comment (yoav-lavi) [reply README.md]: Done

**Pull request merged**